packages/vite/src/node/optimizer/esbuildDepPlugin.ts

```diff
@@ -59,6 +59,10 @@ export function esbuildDepPlugin(
     ? externalTypes.filter((type) => !extensions?.includes('.' + type))
     : externalTypes
 
+  if (!cssBundle) {
+    allExternalTypes.unshift('css')
+  }
```

Comment:
We shouldn't be mutating this array as things will act funny if you build more than once